contracts/core/LyraVault.sol

```diff
@@ -70,14 +67,20 @@ contract LyraVault is Ownable, BaseVault {
   }
 
   /// @dev anyone can trigger a trade
-  function trade() external {
+  function trade(uint strikeId) external {
     require(vaultState.roundInProgress, "round closed");
+
+    uint collateralToAdd = strategy.getRequiredCollateral(strikeId);
+
     // open a short call position on lyra and collect premium
     uint collateralBefore = IERC20(vaultParams.asset).balanceOf(address(this));
+    require(
+      IERC20(vaultParams.asset).transfer(address(strategy), collateralToAdd),
+      "collateral transfer to strategy failed"
+    );
 
-    // todo: detail strategy
     // perform trade through strategy
-    (uint realPremium, uint positionId) = strategy.doTrade();
+    (uint positionId, uint realPremium) = strategy.doTrade(strikeId, collateralToAdd, lyraRewardRecipient);
```

Comment:
I think we can just let the strategy do the following:
1. calculate needed collateral
2. call `transferFrom` to get the collateral
3. trade

Instead of going back and forth here, as the only intermediate step is calling transfer here.

Reply:
I separated these out so that the strategy never has full allowance over the Vault's funds (minimize risk). Do you think that's unnecessary?

Reply:
i feel like the flow is unnecessary, the reason i don't like it is because we put more assumption within the interaction of the 2 contracts, which make it harder to understand: for example these 2 functions will have to be called in sequence, and the caller contract has to make sure they transfer the fund before calling the second function.

If we're really afraid of the unlimited approval, I think it's better that we do one of the following:
1. reset allowance in `setStrategy` when we set a new strategy. To make sure the deprecated strategy contract don't have allowance to move fund.
3. set allowance in beginning of `trade` and reset at the end of the transaction

Reply:
agreed, does make the flow much more confusing. I went with #1 for now.